src/parser.rs

```diff
@@ -63,6 +63,7 @@ fn header(input: &[u8]) -> IResult<&[u8], (&[u8], &[u8])> {
 
 /// Parse a WARC header block.
 // TODO: evaluate the use of `ErrorKind::Verify` here.
+#[allow(clippy::type_complexity)]
```

Comment:
Is this required? What lint is this suppressing?

Reply:
This is suppressing this lint: https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity

I could have also introduced a better new type definition but I honestly wasn't sure enough that I know the domain well enough. Maybe introducing something like `type Headers = (&str, Vec<(&str, &[u8])>, usize)` might be helpful?

Reply:
Thanks for the reference. I'm fine with that for now.